src/transformers/models/llava_next/modeling_llava_next.py

```diff
@@ -65,10 +66,55 @@ def get_anyres_image_grid_shape(image_size, grid_pinpoints, patch_size):
     if not isinstance(grid_pinpoints, list):
         raise ValueError("grid_pinpoints should be a list of tuples or lists")
 
+    # ! VERY IMPORTANT if image_size is tensor, must convert to into tuple, otherwise it will cause wrong calculate
+    if not isinstance(image_size, (list, tuple)):
+        if not isinstance(image_size, (torch.Tensor, np.ndarray)):
+            raise ValueError(
+                f"image_size invalid type: {type(image_size)} not valid, should be either list, tuple, np.ndarray or tensor"
+            )
+        image_size = image_size.tolist()
+
     height, width = select_best_resolution(image_size, grid_pinpoints)
     return height // patch_size, width // patch_size
 
 
+def image_size_to_num_patches(image_size, grid_pinpoints, patch_size: int):
+    """
+    Calculate the number of patches after the preprocessing for images of any resolution.
+
+    Args:
+        image_size (`Union[torch.LongTensor, np.ndarray, Tuple[int, int]):
+            The size of the input image in the format (height, width). ?
+        grid_pinpoints (`List`):
+            A list containing possible resolutions. Each item in the list should be a tuple or list
+            of the form `(height, width)`.
+        patch_size (`int`):
+            The size of each image patch.
+
+    Returns:
+        int: the number of patches
+    """
+    if not isinstance(grid_pinpoints, list):
+        raise ValueError("grid_pinpoints should be a list of tuples or lists")
+
+    # ! VERY IMPORTANT if image_size is tensor, must convert to into tuple, otherwise it will cause wrong calculate
+    if not isinstance(image_size, (list, tuple)):
+        if not isinstance(image_size, (torch.Tensor, np.ndarray)):
+            raise ValueError(f"image_size invalid type {type(image_size)} with value {image_size}")
+        image_size = image_size.tolist()
+
+    best_resolution = select_best_resolution(image_size, grid_pinpoints)
```

Comment:
This makes me think we could write a vecotrized version of this - we should really be needing to map back and forth between torch arrays and lists

Reply:
Sorry I didn't understand what you mean by vectorized version?

Reply:
ahh thanks, i see :)

Reply:
Just had opportunity to look in detail for the `select_best_resolution` and it seems that we never pass in arrays as input. As it's commented in the modeling file, arrays cause wrong calculations for some cases 🤔 

So the question is do we still need to make a vectorized version for `select_best_resolution` or I am not getting the main point

Reply:
Yes, and since we rely on a "loop over tuples" in both cases, I am wondering why we need tensorized version

Reply:
My point is not that we _need_ a vectorized version, but that converting to tuples and the iterating over them is needlessly expensive. A vectorized version is not only faster, but has the advantage of being more compatible in the future for things like `torch.compile`

Reply:
Okay, thanks for clarifying. I think we can leave it for now as a list and loop over it, as most users will probably run a couple of generations.

As for the `compile` there is a community contributor working on it in #29891 and the last comment says that tensorized image sizes are causing compile failures for the contributor. So I would love the "vectorizing + compile + efficiency" to be another PR